code/parachain/frame/composable-maths/src/dex/tests.rs

```diff
@@ -0,0 +1,142 @@
+use crate::dex::constant_product::compute_out_given_in_new;
+use proptest::prelude::*;
+use sp_runtime::Permill;
+
+/// Tests related to constant product math functions
+mod constant_product {
+	use super::*;
+	/// Tests related to the function `compute_out_given_in_new`
+	mod compute_out_given_in_new {
+		use super::*;
+
+		/// should always be equal to the number of elements in the `list` array found in
+		/// `checked_inputs_and_outputs` function
+		const CHECKED_I_AND_O_SIZE: u32 = 5;
+
+		prop_compose! {
+			/// Returns (w_i, w_o, b_i, b_o, a_sent, fee, a_out, fee_out)
+			fn checked_inputs_and_outputs()
+			(x in 0..CHECKED_I_AND_O_SIZE) -> (Permill, Permill, u128, u128, u128, Permill, u128, u128) {
```

Comment:
Another nice proptest would be to check on a generated range of possible values for weights, balances and input and see if they cause unexpected errors.

Reply:
While I agree it would be nice to test this on a larger dataset, random or pseudo-random numbers may lead to a flaky test.

Reply:
does not have to be a random generator, just need some specific ranges to be tested with the combinations possible. But not that critical. Feel free to resolve.

Reply:
f62b0bf1ae1ba0a971de22aea3dbe0c34a82db76